emojivoto-voting-svc/voting/poll.go

```diff
@@ -57,8 +62,20 @@ func (p *inMemoryPoll) Results() ([]*Result, error) {
 	return results, nil
 }
 
+var addCounter sync.Once
+var counter *prometheus.CounterVec
```

Comment:
rather than using a `sync.Once`, could you simply add the counter initialization here?  e.g.

```
var counter *prometheus.CounterVec = promauto.NewCounterVec(...)
```

Reply:
I tried this diff and tests passed for me:

```
diff --git a/emojivoto-voting-svc/voting/poll.go b/emojivoto-voting-svc/voting/poll.go
index cb20a3a..dbb4c5e 100644
--- a/emojivoto-voting-svc/voting/poll.go
+++ b/emojivoto-voting-svc/voting/poll.go
@@ -63,16 +63,12 @@ func (p *inMemoryPoll) Results() ([]*Result, error) {
 }

 var addCounter sync.Once
-var counter *prometheus.CounterVec
-
-func NewPoll() Poll {
-       addCounter.Do(func() {
-               counter = promauto.NewCounterVec(prometheus.CounterOpts{
+var counter *prometheus.CounterVec = promauto.NewCounterVec(prometheus.CounterOpts{
                        Name: "emojivoto_votes_total",
                        Help: "Number of emoji votes",
                }, []string{"emoji"})
-       })

+func NewPoll() Poll {
        poll := &inMemoryPoll{
                votes:   make(map[string]int, 0),
                counter: counter,
```